pkg/controller/networkpolicy/validate.go

```diff
@@ -760,6 +764,17 @@ func (v *antreaPolicyValidator) validateMulticastIGMP(ingressRules, egressRules
 	return "", true
 }
 
+// validateL7Protocols validates the action field is only set as Allow when L7Protocols
+// field is set in Antrea-native policy rules.
+func (v *antreaPolicyValidator) validateL7Protocols(ingressRules, egressRules []crdv1alpha1.Rule) (string, bool) {
+	for _, r := range append(ingressRules, egressRules...) {
+		if len(r.L7Protocols) > 0 && *r.Action != crdv1alpha1.RuleActionAllow {
+			return "layer 7 protocols can only be used with Allow action", false
+		}
+	}
+	return "", true
```

Comment:
Could you also add a check: when HTTP is set, L4 protocol is either empty or TCP, and "protocols" (icmp and igmp) cannot be set?

Reply:
For sure, done.